### PR TITLE
[MLIR] Move warp_execute_on_lane_0 from vector to gpu

### PR DESCRIPTION
**Pull request by kurapov-peter**
Please see the related RFC here: https://discourse.llvm.org/t/rfc-move-execute-on-lane-0-from-vector-to-gpu-dialect/82989.

This patch does exactly one thing - moves the op to gpu.

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
LGTM.

Some bits could use cleanup, but keeping it to pure code motion makes this PR much easier to review and land for sure.

**Review by Groverkss — APPROVED** [review]
LGTM, Please wait for a day or two before landing so others can review as well

File: mlir/include/mlir/Dialect/GPU/IR/GPUOps.td
Comment (kurapov-peter) [reply mlir/include/mlir/Dialect/GPU/IR/GPUOps.td:1103]: Maybe? I had to add it to resolve some missing default constructors, and used the same approach as in `GPU_ReturnOp` and `Vector_YieldOp`, assuming they are empty for a reason. Looking at it now, it should be equivalent, but you're right. I think that would be the correct way of doing it to not have problems if things change. I can submit a patch for all such cases separately.

File: mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp
Comment (banach-space) [inline mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp:1]: Shouldn't this file also be moved to GPU?

Comment (kurapov-peter) [reply mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp:1]: Yep, I planned to do that separately, so that this is only the op move

Comment (banach-space) [reply mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp:1]: Makes sense.

Don't wait for my approval, this makes sense, but is also outside my area of expertise and you already have +1 from two folks active in this area.

**Comment by kurapov-peter** [comment]
There seem to be no objections (both on the PR and RFC), so I'm landing this shortly.

**Pull request merged**
